backstage/packages/backend/src/index.ts

```diff
@@ -4,6 +4,96 @@ import { githubOrgEntityProviderTransformsExtensionPoint } from '@backstage/plug
 // Import { legacyPlugin } from '@backstage/backend-common';
 import { myVerifiedUserTransformer } from './transformers';
 
+// Sign in users who are not in the catalog
+import { googleAuthenticator } from '@backstage/plugin-auth-backend-module-google-provider';
+import { githubAuthenticator } from '@backstage/plugin-auth-backend-module-github-provider';
+import {
+    stringifyEntityRef,
+    DEFAULT_NAMESPACE,
+} from '@backstage/catalog-model';
+import {
+    authProvidersExtensionPoint,
+    createOAuthProviderFactory,
+} from '@backstage/plugin-auth-node';
+import { NotFoundError } from '@backstage/errors';
+
+const customAuth = createBackendModule({
+    // This ID must be exactly "auth" because that's the plugin it targets
+    pluginId: 'auth',
+    // This ID must be unique, but can be anything
+    moduleId: 'custom-google-auth-provider', // todo: change this to something more descriptive
+    register(reg) {
+        reg.registerInit({
+            deps: {
+                providers: authProvidersExtensionPoint,
+            },
+            async init({ providers }) {
+                providers.registerProvider({
+                    // This ID must match the actual provider config, e.g. addressing
+                    // auth.providers.github means that this must be "github".
+                    providerId: 'google',
+                    // Use createProxyAuthProviderFactory instead if it's one of the proxy
+                    // based providers rather than an OAuth based one
+                    factory: createOAuthProviderFactory({
```

Comment:
This feels like a lot of nested logic crammed into a single function call. Maybe break the factory call out? It doesn't look like you're using any of the existing context anyways.

Reply:
I think I want leave it this way for now, this way it'll match the examples in the docs, so I'll have something to reference if I need to troubleshoot